include/revng/Pipes/Ranks.h

```diff
@@ -4,21 +4,23 @@
 // This file is distributed under the MIT License. See LICENSE.md for details.
 //
 
+#include "revng/Model/Function.h"
 #include "revng/Model/Type.h"
 #include "revng/Pipeline/Rank.h"
 #include "revng/Support/MetaAddress.h"
+#include "revng/Support/YAMLTraits.h"
 
 namespace revng::pipes::ranks {
 
 inline auto Binary = pipeline::defineRootRank<"binary">();
 
 using pipeline::defineRank;
-inline auto Function = defineRank<"function", MetaAddress>(Binary);
+inline auto Function = defineRank<"function", model::Function::Key>(Binary);
 inline auto BasicBlock = defineRank<"basic-block", MetaAddress>(Function);
 inline auto Instruction = defineRank<"instruction", MetaAddress>(BasicBlock);
 
 inline auto Type = defineRank<"type", model::Type::Key>(Binary);
-inline auto TypeField = defineRank<"type-field", uint64_t>(Type);
+inline auto TypeField = defineRank<"type-field", std::tuple<uint64_t>>(Type);
```

Comment:
Why do we have a `std::tuple<uint64_t>` again? It makes me quite sad.

Reply:
To make it work with e.g. `StructField.key()`